### PR TITLE
ARTabletop setup

### PR DESCRIPTION
**Pull request by doerrdude**
Added ARTabletop setup folder which contains all the logic, materials, scripts, shader and a prefab to run the ARTabletop demo. Also added ARTabletopScene to MapboxARScenes folder.

@atripathi-mb can you look at these and do a review before merging?

### CODE REVIEW

File: Assets/ARTabletop/Scripts/PlaceMapOnARPlane.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using UnityEngine;
+using UnityARInterface;
+
+public class PlaceMapOnARPlane : MonoBehaviour {
+
+    [SerializeField]
+    Transform _mapTransform;
+
+	// Use this for initialization
+	void Start ()
```
Comment (abhishektrip) [inline Assets/ARTabletop/Scripts/PlaceMapOnARPlane.cs]: Fix formatting.

File: Assets/ARTabletop/Shader/Clip_Box.shader [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+//    SOFTWARE.
+
+// clips any pixel in an area defined by size, rotation, and position
+
+Shader "Clip/Box" {
+	Properties {
+    
+      //_BoxSize ("Box Size (World Space)", Vector) = (1,1,1,1)
+      //_BoxRotation("Box Rotation (Euler xyz)", Vector) = (0,0,0,1)
+      //_Origin ("Origin (World Space)", Vector) = (0,0,0,0)
```
Comment (abhishektrip) [inline Assets/ARTabletop/Shader/Clip_Box.shader]: Remove unused code.

File: Assets/ARTabletop/Shader/Clip_Box.shader [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      Tags { "RenderType" = "Opaque" }
+      Cull Off
+      CGPROGRAM
+      // Physically based Standard lighting model, and enable shadows on all light types
+	  #pragma surface surf Standard fullforwardshadows
+
+	  // Use shader model 3.0 target, to get nicer looking lighting
+	  #pragma target 3.0
+		
+      struct Input {
```
Comment (abhishektrip) [inline Assets/ARTabletop/Shader/Clip_Box.shader]: Fix formatting. Use brackets in new line.

File: Assets/ARTabletop/Shader/Clip_Box.shader [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+          float2 uv_MainTex;
+          float2 uv_Conversion;
+          float2 uv_BumpMap;
+          float3 worldPos;
+      };
+      
+      half _Glossiness;
```
Comment (abhishektrip) [inline Assets/ARTabletop/Shader/Clip_Box.shader]: fix indentation

File: Assets/ARTabletop/Shader/Clip_Box.shader [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+      float3 _Origin;
+      float3 _BoxSize;
+      float3 _BoxRotation;
+      
+      sampler2D _MainTex;
+      sampler2D _BumpMap;
+      sampler2D _Conversion;
+      
+      void surf (Input IN, inout SurfaceOutputStandard o) {
```
Comment (abhishektrip) [inline Assets/ARTabletop/Shader/Clip_Box.shader]: Bracket in next line.

File: Assets/ARTabletop/Shader/Clip_Box.shader [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+	      // Rodrigues' rotation formula uses radians
+		  // Vector3 newDir = Mathf.Cos(angle) * dir + Mathf.Sin(angle) * Vector3.Cross(axis, dir) + (1.0f - Mathf.Cos(angle)) * Vector3.Dot(axis,dir) * axis;
```
Comment (abhishektrip) [inline Assets/ARTabletop/Shader/Clip_Box.shader]: remove unused code.

File: Assets/ARTabletop/Shader/Clip_Box.shader [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+	  	  float3 dir = IN.worldPos - _Origin;
+	  	  //dir = normalize(dir);
```
Comment (abhishektrip) [inline Assets/ARTabletop/Shader/Clip_Box.shader]: remove unused code.

File: Assets/ARTabletop/Shader/Clip_Box.shader [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+	      	  abs(dir.z)  // no negatives
+      	  );
+
+          // not good to use if statements in shaders, not sure about this EDIT: don't use this either
+//        clip( 
+//	      	(dist.x - _BoxSize.x * 0.5 < 0) &&
+//	      	(dist.y - _BoxSize.y * 0.5 < 0) &&
+//	      	(dist.z - _BoxSize.z * 0.5 < 0)
+//        ? -1:1);
+ 		  // replaced
```
Comment (abhishektrip) [inline Assets/ARTabletop/Shader/Clip_Box.shader]: remove unused code.

File: Assets/ARTabletop/Shader/Clip_Box.shader [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+		  dist.x = dist.x - _BoxSize.x * 0.5;
+		  dist.y = dist.y - _BoxSize.y * 0.5;
+		  dist.z = dist.z - _BoxSize.z * 0.5;
+
+      	  // all need to be less than size
```
Comment (abhishektrip) [inline Assets/ARTabletop/Shader/Clip_Box.shader]: Fix indentation

File: Assets/ARTabletop/Shader/Clip_Box.shader [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+      	  half t = min(1, dist.x); // if greater than zero don't clip
+      	  t = max(t, dist.y);
+      	  t = max(t, dist.z);
+
+
+      	  clip(-1*t);
+      	  //clip(-1*(step(dist.x,0) && step(dist.y,0) && step(dist.z,0)));
+
+      	  //clip(t);
```
Comment (abhishektrip) [inline Assets/ARTabletop/Shader/Clip_Box.shader]: remove unused code.

File: Assets/ARTabletop/Shader/Clip_Box.shader [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+          // min = 0 // value = dist // max = _ConvertDistance
```
Comment (abhishektrip) [inline Assets/ARTabletop/Shader/Clip_Box.shader]: remove unused code.

File: Assets/ARTabletop/Scripts/PlaceMapOnARPlane.cs [hunk 12]
```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityARInterface;
+
+public class PlaceMapOnARPlane : MonoBehaviour {
```
Comment (abhishektrip) [inline Assets/ARTabletop/Scripts/PlaceMapOnARPlane.cs]: Brackets in the next line please.

File: Assets/ARTabletop/Scripts/ARTableTileProvider.cs [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+			}
+
+		}
+
+		private void Update()
+		{
+            
+			if (!_initialized) return;
+
+            _currentTile = TileCover.CoordinateToTileId(_basicMap.WorldToGeoPosition(_targetTransform.position),(int)_map.Zoom);
```
Comment (abhishektrip) [inline Assets/ARTabletop/Scripts/ARTableTileProvider.cs]: indentation.

File: Assets/ARTabletop/Scripts/ARTableTileProvider.cs [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+		public override void OnInitialized()
+		{
+			if (_targetTransform == null)
+			{
+				Debug.LogError("TransformTileProvider: No location marker transform specified.");
+				Destroy(this);
+			}
+			else
+			{
+                _initialized = true;
```
Comment (abhishektrip) [inline Assets/ARTabletop/Scripts/ARTableTileProvider.cs]: indentation

File: Assets/ARTabletop/Scripts/ARTableTileProvider.cs [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+	using Mapbox.Map;
+	using Mapbox.Utils;
+	using Mapbox.Unity.Utilities;
+    using System.Collections.Generic;
+
+	public class ARTableTileProvider : AbstractTileProvider
+	{
+        
+		[SerializeField]
+        private AbstractMap _basicMap;
```
Comment (abhishektrip) [inline Assets/ARTabletop/Scripts/ARTableTileProvider.cs]: indentation

File: Assets/ARTabletop/Scripts/ARTableTileProvider.cs [hunk 16]
```diff
@@ -0,0 +1,83 @@
+﻿namespace Mapbox.Unity.Map
+{
+	using System.Linq;
+	using UnityEngine;
+	using Mapbox.Map;
+	using Mapbox.Utils;
+	using Mapbox.Unity.Utilities;
+    using System.Collections.Generic;
```
Comment (abhishektrip) [inline Assets/ARTabletop/Scripts/ARTableTileProvider.cs]: indentation

**Review by abhishektrip — CHANGES_REQUESTED** [review]
Logic looks good 👍

File: Assets/ARTabletop/Scripts/ARPlaneHandler.cs [hunk 17]
```diff
@@ -0,0 +1,35 @@
+﻿using UnityEngine;
```
Comment (wilhelmberg) [inline Assets/ARTabletop/Scripts/ARPlaneHandler.cs:1]: Whole file has spaces, please use tabs.

File: Assets/ARTabletop/Scripts/ARTableTileProvider.cs [hunk 18]
```diff
@@ -0,0 +1,80 @@
+﻿using System.Linq;
```
Comment (wilhelmberg) [inline Assets/ARTabletop/Scripts/ARTableTileProvider.cs:1]: Whole file has mixed tabs and spaces.

File: Assets/ARTabletop/Scripts/PlaceMapOnARPlane.cs [hunk 19]
```diff
@@ -0,0 +1,35 @@
+﻿using UnityEngine;
```
Comment (wilhelmberg) [inline Assets/ARTabletop/Scripts/PlaceMapOnARPlane.cs:1]: Whole file has spaces.

File: Assets/ARTabletop/Scripts/UpdateShaderCoordinatesByARPlane.cs [hunk 20]
```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections;
```
Comment (wilhelmberg) [inline Assets/ARTabletop/Scripts/UpdateShaderCoordinatesByARPlane.cs:1]: Whole file has spaces.

File: Assets/ARTabletop/Shader/Clip_Box.shader [hunk 21]
```diff
@@ -0,0 +1,125 @@
+﻿
```
Comment (wilhelmberg) [inline Assets/ARTabletop/Shader/Clip_Box.shader:1]: Whole file has mixed tabs and spaces.

**Review by wilhelmberg — CHANGES_REQUESTED** [review]
@lauri-rustanius still some formatting issues.

**Review by wilhelmberg — APPROVED** [review]

**Comment by wilhelmberg** [comment]
@lauri-rustanius 
I took the liberty of fixing the remaining formatting issues so that your are free to merge from my point of view.

**Pull request merged**